backend/tracim_backend/lib/translate/providers.py

```diff
@@ -37,15 +40,63 @@ def __init__(
         self._session = session
         self._config = config
 
-    def get_translation_service(self) -> typing.Optional[TranslationService]:
+    def get_translation_service(self) -> TranslationService:
 
         if self._config.TRANSLATION_SERVICE__PROVIDER == TranslationProvider.SYSTRAN:
             return SystranTranslationService(
                 api_url=self._config.TRANSLATION_SERVICE__SYSTRAN__API_URL,
                 api_key=self._config.TRANSLATION_SERVICE__SYSTRAN__API_KEY,
+                timeout=self._config.TRANSLATION_SERVICE__TIMEOUT,
             )
         elif self._config.TRANSLATION_SERVICE__PROVIDER == TranslationProvider.TEST:
             logger.warning(self, "Running in test translation service !")
             return TestTranslationService()
         else:
             raise TracimException("Translation Service not available")
+
+    def translate_raw_content(
```

Comment:
I don't really understood the separation between translate_raw_content usage and view for now.
`get_html_document_translation` do not use this method and `filename` is modified both outside and inside the function which seems confusing for me.

Reply:
Ah… yes I forgot to use it in `get_html_document_translation`, this is fixed.